### PR TITLE
[FR] rust translation chap7 end

### PR DESCRIPTION
**Pull request by mgemard**
Finished chapter 7 of tour of rust.

### CODE REVIEW

**Comment by mgemard** [comment]
I kept the resources you give in the conclusion of chapter 7 even though they are in english.

I said something like this:

```
Bien que les ressources suivantes soient en anglais, nous vous recommendons
        celles-ci. Car de toute façon il faudra bien se mettre à l'anglais,
        pas vrai <span class=\"emoji\">\U0001F604</span>?
```

File: lessons.yaml [hunk 1]
```diff
@@ -5371,6 +5371,21 @@ pages:
         * `generics` - Using parameterized types effectively create struct/functions known types and thus known sizes.
         * `indirection` - Putting instances on the heap gives us a level of indirection that allow us to not have to worry about the size of the actual type and just store a pointer to it. There are other ways as well!
 
+    fr:
+      title: Gestion des données sans Taille
+      content_markdown: |
+        Les traits présentent un défi intéressant lorsque nous voulons les stocker
+        dans une autre structure. Les traits masquent la structure originel et
+        notamment sa taille. Avec Rust, les éléments sans taille connues et qui sont
+        stockées dans une structure sont traitées de deux manières:
```
Comment (VincentFoulon80) [inline lessons.yaml:5380]:
```suggestion
        Les traits présentent un défi intéressant lorsque nous voulons les stocker
        dans une autre structure. Les traits masquent la structure originelle et
        notamment sa taille. Avec Rust, les éléments sans taille connue et qui sont
        stockées dans une structure sont traitées de deux manières:
```

File: lessons.yaml [hunk 2]
```diff
@@ -5399,6 +5414,30 @@ pages:
...
+      content_markdown: |
+        Avec Rust les génériques vont de pair avec les traits. Lorsque qu'un paramètre
+        de fonctions est un type paramétré `T`, le compilateur contraint les types qui
+        peuvent être utilisés comme argument de la fonction. Le mot-clé `where` permet
+        de lister les traits que l'argument doit implémenter.
```
Comment (VincentFoulon80) [inline lessons.yaml:5424]:
```suggestion
        Avec Rust les génériques vont de pair avec les traits. Lorsqu'un paramètre
        de fonction est un type paramétré `T`, le compilateur contraint les types qui
        peuvent être utilisés comme argument de la fonction. Le mot-clé `where` permet
        de lister les traits que l'argument doit implémenter.
```

File: lessons.yaml [hunk 3]
```diff
@@ -5479,6 +5542,26 @@ pages:
...
+      content_markdown: |
+        `Box` est une structure de données qui permet de déplacer les données
+        de la pile vers le tas.
+
+        `Box` est également appelé *pointeur intelligent* (en anglais smart
+        pointer) qui contient un pointeur vers des données sur le tas.
+
+        `Box` étant de taille connue (la taille du pointeur), on l'utilise
+        souvant comme un moyen de stocker une référence dans une structure
+        lorsque cette dernière doit connaître la taille de certaints éléments.
```
Comment (VincentFoulon80) [inline lessons.yaml:5557]:
```suggestion
        `Box` étant de taille connue (la taille du pointeur), on l'utilise
        souvent comme un moyen de stocker une référence dans une structure
        lorsque cette dernière doit connaître la taille de certains éléments.
```

File: lessons.yaml [hunk 4]
```diff
@@ -5511,6 +5594,30 @@ pages:
...
+        where
+            T: MyTrait
+        {
+            foo: T
+            ...
+        }
+        ```
+        Quand à la définition des méthodes d'une structure paramétrée, celle-ci
+        est faîte de la manière suivante (le type paramétré peut être utilisé
+        dans le bloc d'implémentation):
```
Comment (VincentFoulon80) [inline lessons.yaml:5619]:
```suggestion
        Quant à la définition des méthodes d'une structure paramétrée, celle-ci
        est faite de la manière suivante (le type paramétré peut être utilisé
        dans le bloc d'implémentation):
```

File: lessons.yaml [hunk 5]
```diff
@@ -5552,6 +5659,22 @@ pages:
...
+    fr:
+      title: Chapitre 7 - Conclusion
+      content_markdown: |
+        Nous en savons désormais beaucoup pour pouvoir représenter clairement
+        nos idées avec Rust! Les abstractions peuvent sembler simples mais elles
+        sont suffisamment puissantes pour faire en sorte qu'écrirer un programme
+        Rust devienne une partie de plaisir. Dans ce chapitre, nous avons vu très
+        brièvement les pointeurs intelligents avec `Box`. Dans le chapitre suivant,
+        nous découvrirons comment les pointeurs intelligents peuvent nous aider
+        dans d'autre cas où une gestion spéciale de la mémoire est requise.
```
Comment (VincentFoulon80) [inline lessons.yaml:5671]:
```suggestion
        Nous en savons désormais beaucoup pour pouvoir représenter clairement
        nos idées avec Rust! Les abstractions peuvent sembler simples mais elles
        sont suffisamment puissantes pour faire en sorte qu'écrire un programme
        Rust devienne une partie de plaisir. Dans ce chapitre, nous avons vu très
        brièvement les pointeurs intelligents avec `Box`. Dans le chapitre suivant,
        nous découvrirons comment les pointeurs intelligents peuvent nous aider
        dans d'autre cas où une gestion spéciale de la mémoire est requise.
```

**Comment by richardanaya** [comment]
Awesome as always gents

**Pull request merged**